manticore/core/smtlib/solver.py

```diff
@@ -735,18 +770,162 @@ def __init__(self):
 
 class CVC4Solver(SMTLIBSolver):
     def __init__(self):
-        init = ["(set-logic QF_AUFBV)", "(set-option :produce-models true)"]
-        command = f"{consts.cvc4_bin} --tlimit={consts.timeout * 1000} --lang=smt2 --incremental"
+        init = info.inits["cvc4"]
+        command = info.commands["cvc4"]
+        self.ncores = 1
         super().__init__(command=command, init=init)
 
 
 class BoolectorSolver(SMTLIBSolver):
-    def __init__(self):
-        init = ["(set-logic QF_AUFBV)", "(set-option :produce-models true)"]
-        command = f"{consts.boolector_bin} --time={consts.timeout} -i"
+    def __init__(self, args: List[str] = []):
+        init = info.inits["boolector"]
+        command = info.commands["boolector"]
+        self.ncores = 1
         super().__init__(command=command, init=init)
 
 
+class SmtlibPortfolio:
+    def __init__(self, solvers: List[str], debug: bool = False):
```

Comment:
`SMTLIBSolver` instead of `str`?

Reply:
We keep a list of solver names in several dicts.